runtime/near-vm-runner/src/tests/test_builder.rs

```diff
@@ -0,0 +1,188 @@
+        self
+    }
+
+    pub(crate) fn expect(self, want: expect_test::Expect) {
+        self.expects(&[want])
+    }
+
+    pub(crate) fn expects(self, wants: &[expect_test::Expect]) {
+        let runtime_config_store = RuntimeConfigStore::new(None);
+
+        let mut protocol_versions = Vec::new();
+        for feat in self.protocol_features {
+            protocol_versions.push(feat.protocol_version() - 1)
+        }
+        protocol_versions.push(PROTOCOL_VERSION);
+        assert_eq!(
+            wants.len(),
+            protocol_versions.len(),
+            "specified {} features but only {} expectation ({} + 1 needed)",
+            self.protocol_features.len(),
+            wants.len(),
+            self.protocol_features.len()
+        );
+
+        for (want, &protocol_version) in wants.iter().zip(&protocol_versions) {
+            let mut results = vec![];
+            for vm_kind in [VMKind::Wasmer2, VMKind::Wasmer0, VMKind::Wasmtime] {
+                if self.skip.contains(&vm_kind) {
+                    continue;
+                }
+                let runtime_config = runtime_config_store.get_config(protocol_version);
+
+                let mut fake_external = MockedExternal::new();
+                let config = runtime_config.wasm_config.clone();
+                let fees = RuntimeFeesConfig::test();
+                let context = self.context.clone();
+
+                let promise_results = vec![];
+
+                let runtime = vm_kind.runtime(config).expect("runtime has not been compiled");
+                let res = runtime.run(
+                    &self.code,
+                    &self.method,
+                    &mut fake_external,
+                    context,
+                    &fees,
+                    &promise_results,
+                    protocol_version,
+                    None,
+                );
+
+                let mut got = if self.opaque_outcome {
+                    String::new()
+                } else {
+                    format!("{:?}\n", res.outcome())
+                };
+                if let Some(err) = res.error() {
+                    let mut err = err.to_string();
+                    assert!(err.len() < 1000, "errors should be bounded in size");
+                    if self.opaque_error {
+                        err = "...".to_string();
+                    }
+                    writeln!(got, "Err: {err}").unwrap();
+                }
+
+                results.push((vm_kind, got));
+            }
+
+            if !results.is_empty() {
+                want.assert_eq(&results[0].1);
+                for i in 1..results.len() {
+                    if results[i].1 != results[0].1 {
+                        panic!(
+                            "Inconsistent VM Output:\n{:?}:\n{}\n\n{:?}:\n{}",
+                            results[0].0, results[0].1, results[i].0, results[i].1
+                        )
+                    }
+                }
```

Comment:
Ther's a benefit in saying that the outcome is inconsistent, rather than not matching the template. The second would just flip-flob between the first two mismatching outputs (*)

(*) actually, there's a bug in expect-test where using the same `expcet![]` twice produces garbage during update, due to the beloved aliasing issues. I probably should fix that upstream, but "specific error about inconsistency is good in itself" is a nice way to sweep this under the rug for the time being.